applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -256,6 +255,141 @@ static struct bt_le_per_adv_sync_cb sync_callbacks = {
 	.term = pa_sync_terminated_cb,
 };
 
+static void pa_timer_handler(struct k_work *work)
+{
+	int ret;
+
+	if (req_recv_state != NULL) {
+		enum bt_bap_pa_state pa_state;
+
+		if (req_recv_state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
+			pa_state = BT_BAP_PA_STATE_NO_PAST;
+		} else {
+			pa_state = BT_BAP_PA_STATE_FAILED;
+		}
+
+		ret = bt_bap_scan_delegator_set_pa_state(req_recv_state->src_id, pa_state);
+		if (ret) {
+			LOG_ERR("set PA state to %d failed, err = %d", pa_state, ret);
+		}
+	}
+}
+
+static K_WORK_DELAYABLE_DEFINE(pa_timer, pa_timer_handler);
+
+static int pa_sync_past(struct bt_conn *conn, uint16_t pa_interval)
+{
+	int ret;
+	struct bt_le_per_adv_sync_transfer_param param = {0};
+
+	param.skip = PA_SYNC_SKIP;
+	param.timeout = interval_to_sync_timeout(pa_interval);
+
+	ret = bt_le_per_adv_sync_transfer_subscribe(conn, &param);
+	if (ret != 0) {
+		LOG_WRN("Could not do PAST subscribe: %d", ret);
+	} else {
+		LOG_DBG("Syncing with PAST: %d", ret);
+		(void)k_work_reschedule(&pa_timer, K_MSEC(param.timeout * 10));
+	}
+
+	return ret;
```

Comment:
This has not been changed or answered

Reply:
I check with Emil and this is the easy way to prevent checking if there's underlying work.
From what I seen from log, this won't be called multiple times.